File: app/src/main/java/com/islam/basepropject/project_base/utils/LocalManager.java
package com.islam.basepropject.project_base.utils;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Build;

import androidx.preference.PreferenceManager;

import java.util.Locale;

public class LocalManager {

    public static void setLocale(Context c) {
        setNewLocale(c, getLanguage(c));
    }

    public static void setNewLocale(Context c, String language) {
        persistLanguage(c, language);
        updateResources(c, language);
    }

    public static boolean isCurrentLanguage(Context c,String lang) {
        return getLanguage(c).equalsIgnoreCase(lang);
    }

    public static String getLanguage(Context c) {
        return PreferenceManager.getDefaultSharedPreferences(c).getString("language","en");
    }

    private static void persistLanguage(Context c, String language) {
       // AuthSharedDataSource.setShareLocale(c, language);
    }

    public static Context updateResources(Context context, String language) {
        Locale locale = new Locale(language);
        Locale.setDefault(locale);

        Resources res = context.getResources();
        Configuration config = new Configuration(res.getConfiguration());
        if (Build.VERSION.SDK_INT >= 17) {
            config.setLocale(locale);
            context = context.createConfigurationContext(config);
        } else {
            config.locale = locale;
            res.updateConfiguration(config, res.getDisplayMetrics());

        }
        return context;
    }
}

File: app/src/main/java/com/islam/basepropject/project_base/utils/LocalManager.kt
package com.islam.basepropject.project_base.utils

import android.content.Context
import android.content.res.Configuration
import androidx.preference.PreferenceManager
import java.util.*

object LocalManager {

    fun setLocale(c: Context) {
        setNewLocale(c, getLanguage(c))
    }

    fun setNewLocale(c: Context, language: String?) {
        persistLanguage(c, language)
        updateResources(c, language)
    }

    fun isCurrentLanguage(c: Context, lang: String): Boolean {
        return getLanguage(c)!!.equals(lang, ignoreCase = true)
    }

    fun getLanguage(c: Context): String? {
        return PreferenceManager.getDefaultSharedPreferences(c).getString("language", "en")
    }

    private fun persistLanguage(c: Context, language: String?) {
        PreferenceManager.getDefaultSharedPreferences(c).edit().putString("language", language).apply()
    }

    fun updateResources(context: Context, language: String?): Context {
        var context = context
        val locale = Locale(language)
        Locale.setDefault(locale)

        val res = context.resources
        val config = Configuration(res.configuration)

        config.setLocale(locale)
        context = context.createConfigurationContext(config)

        return context
    }
}
